### PR TITLE
Changes to align to FragmentSet; Replace ErrorLog with Reflection RecordError

### LINKED ISSUE #217: Align to FragmentSet change
#### Description:
Considering:
BHoM/BHoM#574 (main BHoM.oM change)
BHoM/BHoM_Engine#1248 (aligns BHoM_Engine)

Align this Toolkit to FragmentSet change.

### LINKED ISSUE #218: TAS_Toolkit: remove the ErrorLog and use Reflection.RecordError instead
As part of https://github.com/BHoM/BHoM_Adapter/issues/113
TAS_Toolkitstill uses the old ErrorLog in a few methods.

#### Suggestions to restore compliance:

Replace `ErrorLog.Add` with  `Engine.Reflection.Compute.RecordError`.

### PR DESCRIPTION
**Pull request by alelom**
### NOTE: Depends on 

1. BHoM/BHoM#574 (main BHoM.oM change)
2. BHoM/BHoM_Engine#1248 (aligns BHoM_Engine to 1.)

 ### Issues addressed by this PR

Closes #217 
Closes #218

 

 ### Changelog

- Changes to align to FragmentSet change.
- Replaced all `ErrorLog.Add` with `Engine.Reflection.Compute.RecordError`.

## Important note
To be merged with https://github.com/BHoM/BHoM_Engine/pull/1248

### CODE REVIEW

File: TAS_TBDAdapter/CRUD/Read.cs [hunk 1]
```diff
@@ -209,7 +209,7 @@ public List<Panel> ReadBuildingElements(List<string> ids = null)
                     {
                         Opening newOpening = new Opening();
                         newOpening.Edges = frame.ExternalEdges;
-                        newOpening.Fragments = new List<IBHoMFragment>(pane.Fragments);
+                        newOpening.Fragments = new FragmentSet(pane.Fragments);
```
Comment (alelom) [inline TAS_TBDAdapter/CRUD/Read.cs]: This is the only change needed for FragmentSet (#217 )

File: TAS_TBDAdapter/TASTBDAdapter.cs [hunk 2]
```diff
@@ -101,7 +101,7 @@ public override IEnumerable<object> Pull(IRequest request, Dictionary<string, ob
             }
             catch (Exception e)
             {
-                ErrorLog.Add(e.ToString());
+               Engine.Reflection.Compute.RecordError(e.ToString());
```
Comment (FraserGreenroyd) [inline TAS_TBDAdapter/TASTBDAdapter.cs]: The indentation on this line doesn't look quite right.

File: TAS_T3DAdapter/TAST3DAdapter.cs [hunk 3]
```diff
@@ -134,7 +134,7 @@ public override IEnumerable<object> Pull(IRequest request, Dictionary<string, ob
             }
             catch (Exception e)
             {
-                ErrorLog.Add(e.ToString());
+               Engine.Reflection.Compute.RecordError(e.ToString());
```
Comment (FraserGreenroyd) [inline TAS_T3DAdapter/TAST3DAdapter.cs]: For consistency with other BEnv toolkits and TAS Toolkit itself, could we append `BH.` to the front of all the `RecordError` messages.

**Comment by FraserGreenroyd** [comment]
Added do not merge status until dependent PRs are merged.

**Pull request merged**